src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -202,16 +204,20 @@ def set_opts(self, curl, url, params, headers,
         curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
         curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
 
-        # also accepts encoding/compression algorithms
-        if headers and headers.get("Accept-Encoding"):
-            if isinstance(headers["Accept-Encoding"], basestring):
-                curl.setopt(pycurl.ENCODING, headers["Accept-Encoding"])
+        # If ACCEPT_ENCODING is set to the encoding string, then libcurl
+        # will automatically decode the response object according to the
+        # Content-Enconding received
+        # More info: https://curl.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html
+        thisHeaders = copy.deepcopy(headers)
+        if thisHeaders and thisHeaders.get("Accept-Encoding"):
+            if isinstance(thisHeaders["Accept-Encoding"], basestring):
+                curl.setopt(pycurl.ACCEPT_ENCODING, thisHeaders.pop("Accept-Encoding"))
             else:
                 logging.warning("Wrong data type for header 'Accept-Encoding': %s",
-                                type(headers["Accept-Encoding"]))
+                                type(thisHeaders["Accept-Encoding"]))
         else:
             # add gzip encoding by default
-            curl.setopt(pycurl.ENCODING, 'gzip')
+            curl.setopt(pycurl.ACCEPT_ENCODING, 'gzip')
```

Comment:
I would say it have been kind of a long shot to enforce a `gzip` encoding by default in the previous version. Using  `accept_enoding` as the default header and letting curl to decide I agree is the better way to go for the defaults.

Reply:
I would agree, but honestly I was not aware that pycurl library is capable of dealing with encoding. If we knew this earlier I'll structure my initial gzip PR differently.  The changes seems correct to me.